### PR TITLE
Upgrade protobuf and Generate OpenAPI v3 responseBody content examples

### LINKED ISSUE #241: Examples values are not being generated in OpenApi v3 yaml
When generating OpenAPI v3 yaml, at least requestBody `examples`'s  is not generated.
It works for `example` but not for `examples`.

Example valid yaml according to swagger editor:
```yaml
openapi: 3.0.0
info:
    title: Test API
    version: ""
paths:
    /api/v1/test:
        post:
            tags:
                - Test
            summary: Create a test object
            operationId: postTest
            requestBody:
                content:
                    application/json:
                        schema:
                            type: object
                            properties:
                                campaign_id:
                                    type: string
                        examples:
                          example1:
                            summary: "example 1"
                            value: |-
                              {
                                "campaign_id": "123"
                              }
                          example2:
                            summary: "example 2"
                            value: |-
                              { 
                                "campaign_id": "321"
                              }
  
            responses:
                "200":
                    description: OK
```

When doing a simple parse and generate like this:
```go
doc, err := openapi_v3.ParseDocument([]byte(tmp))
yamlDoc, _ := doc.YAMLValue("")
```

The generated yamlDoc is missing the `examples`'s values and looks like this:
```yaml
openapi: 3.0.0
info:
    title: Test API
    version: ""
paths:
    /api/v1/test:
        post:
            tags:
                - Test
            summary: Create a test object
            operationId: postTest
            requestBody:
                content:
                    application/json:
                        schema:
                            type: object
                            properties:
                                campaign_id:
                                    type: string
                        examples:
                            example1:
                                summary: example 1
                            example2:
                                summary: example 2
            responses:
                "200":
                    description: OK
```

### PR DESCRIPTION
**Pull request by morphar**
Fixes #241

### CODE REVIEW

**Review by timburks — CHANGES_REQUESTED** [review]
Hi, thanks for working through all this! There seem to be three fairly big things here:
1. fixing the go_package names both in the protos and generate_gnostic/main.go
2. fixing the plugin test
3. fixing the missing fields in generated OpenAPI v3 responseBody objects

A few things are making me want to address these issues separately, but if you don't mind making a few changes, I am happy to proceed with this:

- fixing the plugin test involved copying protos into the repo. Could you move those into a new "third_party" directory that you create at the top level? That would just mean having them at "third_party/google/*" instead of "google/*". Later we might automatically import those from their sources, but just moving the files is good for now.

- it looks like a proto formatter was run on the protos that were modified, maybe automatically by your text editor's save function - could we exclude those modifications from this commit? Sorry to ask for that extra work - I see there are some cases (vocabulary.proto for example) where the formatting is helpful, but there are also generated protos (for OpenAPI) that are reformatted and since the formatter isn't in the generation flow, that complicates comparing new generated versions.

- finally, the code that was added to OpenAPIv3.go is a modification to a generated file. I'm happy to take that change in this PR and personally look into fixing the generator (so I'm not requesting any change from you, I just wanted to note this).

Thanks again, Dan!

**Comment by morphar** [comment]
Hi Tim and thanks for the thorough and actionable review.

I will go through you comments, probably later today and update as requested.
With regards to this being a pretty big update: It started out with small changes, but as the protobuf stuff in here is pre the big Go + protobuf update and I work a lot with this, I wasn't able to get to a point, where I could generate AND get the tests to succeed without the protobuf update - probably due to the order of the steps I took.
To be honest I didn't try for too long, as I had already made a test run where I updated all proto related stuff in a fork, which went well.
Considering this, I thought that this library would benefit from the upgrade anyway.
I know it's kind of separate concerns, so if you think it would make more sense, I can try to start the process from scratch with the protobuf as step 1.

With regards to you last point, I'm not sure which code I added, that should be generated - but I haven't looked at it yet.
If I can't find it, I will probably ask you for a pointer in the right direction :)

File: generate-gnostic/main.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func protoOptions(directoryName string, packageName string) []ProtoOption {
 
 		ProtoOption{
 			Name:    "go_package",
-			Value:   directoryName + ";" + packageName,
+			Value:   path.Clean("./"+directoryName) + ";" + packageName,
```
[the remark this answers is not in the document]
Comment (morphar) [reply generate-gnostic/main.go:87]: I'm not entirely sure about this one - it was probably due to me wanting to be on the safe side, before I had tested this part.
I will look into it and probably remove it as I expect things to consistent enough to not need it.

**Comment by timburks** [comment]
@morphar Thanks, just to reiterate something from my earlier comments, I think with some (hopefully minor) adjustments, this set of changes would be ok to make as a group. I guess it's really just two changes (the proto update probably forced the test change), and the change to OpenAPIv3.go can be automated in a followup PR.

**Comment by morphar** [comment]
@timburks sounds good! 👍
I have made the changes you requested, hope it lives up to the standard :)
I did add a small change (1/2 a line) to generate-gnostic/generate-compiler.go, to make the change in OpenAPIv3.go automatic - I hope that's o.k.?

File: generate-gnostic/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (morphar) [reply generate-gnostic/main.go]: I ended up removing it - it was not needed

**Review by timburks — COMMENTED** [review]
Thanks! This looks great - I think your change to the compiler fixed your original problem in a few more places. I just have one small change request to maintain proto compatibility.

File: openapiv3/OpenAPIv3.proto [hunk 2]
```diff
@@ -208,8 +208,8 @@ message Info {
   Contact contact = 4;
   License license = 5;
   string version = 6;
-  repeated NamedAny specification_extension = 7;
-  string summary = 8;
+  string summary = 7;
+  repeated NamedAny specification_extension = 8;
```
[the remark this answers is not in the document]
Comment (morphar) [reply openapiv3/OpenAPIv3.proto]: Of course :) It has been fixed and pushed.

**Review by timburks — APPROVED** [review]
@morphar Thank you!

**Comment by morphar** [comment]
Thank you @timburks! :)
It was a pleasure to contribute 👍 

BTW: have you seen this:
[Async API](https://www.asyncapi.com/)

Would it make sense to add support for it in gnostic?

**Comment by morphar** [comment]
Also: would it make sense to make this a new version (0.5.4). Just ro make it more correct for go.mod ;)

**Comment by timburks** [comment]
I'll push a new tag tomorrow (Pacific Time) - Async API is indeed interesting! Let's discuss it in [this issue](https://github.com/google/gnostic/issues/243).

**Comment by morphar** [comment]
Perfect! Thanks Tim :)

**Pull request merged**